lang/mrustc/Portfile

```diff
@@ -59,46 +62,276 @@ compiler.blacklist *gcc*
+
+# Step 0: let start from building mrustc and its libds
+build.target    -f minicargo.mk bin/mrustc LIBS
+
 destroot {
-    xinstall -m 0755 -W ${worksrcpath} ./bin/${name} ${destroot}${prefix}/bin
-    xinstall -m 0755 -W ${worksrcpath} ./bin/minicargo ${destroot}${prefix}/bin
-    file copy ${worksrcpath}/output-${rust_version} ${destroot}${prefix}/lib/mrustc-${rust_version_major}
+    xinstall -m 0755 ${worksrcpath}/bin/mrustc ${destroot}${prefix}/bin
+
+    # install all .rlib and .rlib.hir to lib/
+    xinstall -d ${destroot}${prefix}/lib/mrustc-${rust_version_major}
+    foreach f [glob -tails -directory ${worksrcpath}/output-${rust_version} lib*.rlib lib*.rlib.hir lib*.rlib.c lib*.rlib.o] {
+        xinstall -m 0644 -W ${worksrcpath}/output-${rust_version} $f \
+            ${destroot}${prefix}/lib/mrustc-${rust_version_major}
+    }
 }
 
-# Run a Hello, world! test
-test.env            RUSTC_VERSION=${rust_version} MRUSTC_LIBDIR=output-${rust_version}
-test.run            yes
-test.target         test
+# Step 1: now I need minicargo
+subport minicargo {
+    description     Alternative implementation of Rust cargo
+    long_description {*}${description}
+
+    build.target    -f minicargo.mk bin/minicargo
+
+    depends_build-append \
+                    port:mrustc
+
+    # prevent rebuild libs
+    post-extract {
+        file mkdir ${worksrcpath}/output-${rust_version}
+        foreach f [glob -directory ${prefix}/lib/mrustc-${rust_version_major} *.rlib] {
+            file copy $f ${worksrcpath}/output-${rust_version}
+        }
+    }
+
+    build.env-append \
+                    MRUSTC=${prefix}/bin/mrustc
+
+    destroot {
+        xinstall -m 0755 ${worksrcpath}/bin/minicargo ${destroot}${prefix}/bin
+    }
+
+    livecheck.type  none
+}
+
+# Step 2: build a cargo by mrustc
+subport cargo-${rust_version_major} {
+    version         ${rust_version}
+    revision        0
+
+    description     Cargo for a rust which was bootstraped via mrustc.
+    long_description {*}${description}
+
+    depends_build-append \
+                    port:minicargo \
+                    port:mrustc
+
+    build.env-append \
+                    MINICARGO=${prefix}/bin/minicargo \
+                    MRUSTC=${prefix}/bin/mrustc
+
+    # prevent rebuild libs
+    post-extract {
+        file mkdir ${worksrcpath}/output-${rust_version}
+        foreach f [glob -directory ${prefix}/lib/mrustc-${rust_version_major} lib*] {
+            file copy $f ${worksrcpath}/output-${rust_version}
+        }
+    }
+
+    build.target    -f minicargo.mk output-${rust_version}/cargo
+
+    destroot {
+        xinstall -m 0755 ${worksrcpath}/output-${rust_version}/cargo ${destroot}${prefix}/bin/cargo-${rust_version_major}
+    }
+
+    livecheck.type  none
+}
+
+# Step 3: build a rustc by mrustc
+subport mrustc-rustc {
+    description     Rust which was compiled by mrustc.
+    long_description {*}${description}
```

Comment:
Maybe mention that this port is intended for MacPorts developers (end users will want either `mrustc` or `rustc-<version>`.)

Reply:
@evanmiller not sure hot to say it short :) Can you suggest anything?

Reply:
@catap Maybe rename it `rustc-bootstrap`? Since it really provides `rustc` and not `mrustc`.

"A stage-1 Rust compiler used only to compile rustc-legacy"